### PR TITLE
[common] Shade kafka-related dependency

### PR DESCRIPTION
**Pull request by fsk119**
# What is the purpose of the change

*Shade Kafka-related dependency.*

### CODE REVIEW

File: flink-sql-connector-mysql-cdc/pom.xml [hunk 1]
```diff
@@ -69,12 +69,14 @@ under the License.
...
-                            <!--                                <relocation>-->
-                            <!--                                    <pattern>org.apache.kafka</pattern>-->
-                            <!--                                    <shadedPattern>com.ververica.cdc.connectors.kafka.shaded.org.apache.kafka</shadedPattern>-->
-                            <!--                                </relocation>-->
-                            <!--                            </relocations>-->
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.kafka</pattern>
+                                    <shadedPattern>
+                                        com.ververica.cdc.connectors.kafka.shaded.org.apache.kafka
```
Comment (wuchong) [inline flink-sql-connector-mysql-cdc/pom.xml:76]:
```suggestion
                                        com.ververica.cdc.connectors.shaded.org.apache.kafka
```

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 2]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
-        }
+    private DeserializationRuntimeConverter convertToBoolean() {
+        return new DeserializationRuntimeConverter() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof Boolean) {
+                    return (Boolean) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:245]: We can remove the `Boolean` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 3]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
-        }
+    private DeserializationRuntimeConverter convertToInt() {
+        return new DeserializationRuntimeConverter() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof Integer) {
+                    return (Integer) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:265]: We can remove the `Integer` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 4]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
+        return new DeserializationRuntimeConverter() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof Integer) {
+                    return ((Integer) dbzObj).longValue();
+                } else if (dbzObj instanceof Long) {
+                    return (long) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:285]: We can remove the `long` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 5]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
+        return new DeserializationRuntimeConverter() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof Float) {
+                    return ((Float) dbzObj).doubleValue();
+                } else if (dbzObj instanceof Double) {
+                    return (double) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:303]: We can remove the `double` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 6]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
-        }
+    private DeserializationRuntimeConverter convertToFloat() {
+        return new DeserializationRuntimeConverter() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof Float) {
+                    return (float) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:319]: We can remove the `float` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 7]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof Long) {
+                    switch (schema.name()) {
+                        case MicroTime.SCHEMA_NAME:
+                            return (int) ((long) dbzObj / 1000);
+                        case NanoTime.SCHEMA_NAME:
+                            return (int) ((long) dbzObj / 1000_000);
+                    }
+                } else if (dbzObj instanceof Integer) {
+                    return (int) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:356]: We can remove the `int` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 8]
```diff
@@ -204,151 +234,249 @@ private DeserializationRuntimeConverter createNotNullConverter(LogicalType type)
...
-        }
+    private DeserializationRuntimeConverter convertToBinary() {
+        return new DeserializationRuntimeConverter() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object convert(Object dbzObj, Schema schema) {
+                if (dbzObj instanceof byte[]) {
+                    return (byte[]) dbzObj;
```
Comment (wuchong) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:435]: We can remove the `byte[]` cast.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java [hunk 9]
```diff
@@ -55,7 +55,7 @@
 import java.time.ZoneId;
 
 /**
- * Deserialization schema from Debezium object to Flink Table/SQL internal data structure {@link
+ * Deserialization schema from `Debezium object to Flink Table/SQL internal data structure {@link
```
Comment (leonardBang) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java:58]: ?

**Review by wuchong — APPROVED** [review]
Thanks @fsk119 for the great work!

**Pull request merged**